pkg/kubelet/kubelet.go

```diff
@@ -2825,28 +2825,43 @@ func (kl *Kubelet) HandlePodSyncs(pods []*v1.Pod) {
 // canResizePod determines if the requested resize is currently feasible.
 // pod should hold the desired (pre-allocated) spec.
 // Returns true if the resize can proceed.
-func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, v1.PodResizeStatus, string) {
+// If the resize can proceed, the returned pod condition is nil. Otherwise,
+// the returned pod condition is of type PodResizePending with the reason
+// why the resize could not proceed.
+func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, *v1.PodCondition) {
 	if v1qos.GetPodQOS(pod) == v1.PodQOSGuaranteed && !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScalingExclusiveCPUs) {
 		if utilfeature.DefaultFeatureGate.Enabled(features.CPUManager) {
 			if kl.containerManager.GetNodeConfig().CPUManagerPolicy == "static" {
 				msg := "Resize is infeasible for Guaranteed Pods alongside CPU Manager static policy"
 				klog.V(3).InfoS(msg, "pod", format.Pod(pod))
-				return false, v1.PodResizeStatusInfeasible, msg
+				kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, msg)
+				return false, &v1.PodCondition{
+					Type:    v1.PodResizePending,
+					Reason:  "Infeasible",
```

Comment:
This should be a constant, probably in the API. https://github.com/kubernetes/kubernetes/blob/d6f61438e01da354b0540b74c0f761751285922d/staging/src/k8s.io/api/core/v1/types.go#L3283

Reply:
done!